atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchTransactionManager.java

```diff
@@ -16,20 +16,13 @@
 
 package com.palantir.atlasdb.keyvalue.api.watch;
 
+import java.util.List;
 import java.util.Optional;
-import java.util.Set;
 
+import com.palantir.atlasdb.transaction.api.TransactionManager;
 import com.palantir.lock.watch.IdentifiedVersion;
-import com.palantir.lock.watch.TransactionsLockWatchEvents;
 
-public interface InternalLockWatchManager extends LockWatchManager {
-    /**
-     * Returns a condensed view of new lock watch events since lastKnownVersion for a set of transactions identified by
-     * their start timestamps.
-     *
-     * @param startTimestamps  a set of start timestamps identifying transactions
-     * @param lastKnownVersion exclusive start version to get events from
-     */
-    TransactionsLockWatchEvents getEventsForTransactions(Set<Long> startTimestamps,
-            Optional<IdentifiedVersion> lastKnownVersion);
+public interface LockWatchTransactionManager extends TransactionManager {
```

Comment:
we've found extending txn manager to be a bit of a folly in the past. Would much appreciate if this method were not an interface method, or were otherwise on TransactionManager. At a minimum, it means you get to pick if SchemaAwareTransactionManager should implement LockWatchTransactionManager _facepalm_

Reply:
Removed